lib/common/models/concerns/maximum_redis_lifetime.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require 'common/models/redis_store'
+
+module Common
+  module MaximumRedisLifetime
+    extend ActiveSupport::Concern
+
+    included do
+      unless self < Common::RedisStore
+        raise ArgumentError, 'Class composing Common::MaximumRedisLifetime must be a Common::RedisStore'
+      end
+
+      extend(ClassMethods)
+
+      attribute :created_at
```

Comment:
this might be useful for all RedisStores?

Reply:
:thinking: How would you see a non-max-lifetime RedisStore using `created_at`?

For `User` and `Session`, `created_at` is only used to calculate if the instance has surpassed it's max lifetime.  For `RedisStore` models that can be extended indefinitely, I'm not sure what need they would have for `created_at`.

Reply:
Mainly to avoid a future dev adding created_at support that collides with the concern. Would be similar to any use case for ActiveRecord that has a created_at. Could always move it later on too.

Reply:
I think you should manually add created_at to the model. And then the mixin should depend on that key. I think that should address Alastair's concern too. No one should be adding the mixin to a model arbitrarily. And we're mixing it in to a subclass of RedisStore, not into RedisStore directly.

Reply:
But yes, I do strongly prefer key declarations being on the model itself, to avoid having to go looking into concerns for how attributes are magically added in.